feat: block quote

chris-cedrone-cengage opened the pull request.

chris-cedrone-cengage committed:
feat: initial commit of the block quote component

chris-cedrone-cengage committed:
Create proud-bulldogs-study.md

orion-cengage reviewed (changes requested):
1. Can we add a space in between the m-dash and the attribution?
![image](https://user-images.githubusercontent.com/24397282/147102637-23d5daaf-9e33-42e4-ae58-5da822b985de.png)

2.  Since the quotation marks are built into the component, can we make sure we're using smart quotes, so they look less like this:
![image](https://user-images.githubusercontent.com/24397282/147102896-957f91f5-92a0-4236-994d-b2f1b884107c.png)

and more like this? 
![image](https://user-images.githubusercontent.com/24397282/147102945-5c86b60b-ad3a-4ab6-9a49-e82f51758dc9.png)

I think it's a matter of using these:
![image](https://user-images.githubusercontent.com/24397282/147103098-9a7bda4f-aa44-498d-87fc-9d5caf98df84.png)

It will be hard to tell with Open Sans, but should be easier to see on Noto Serif.

3. Blockquote is one word, so please fix instances like this:
![image](https://user-images.githubusercontent.com/24397282/147105430-ec231773-bd11-4b8a-83a9-25e34f2bc34b.png)

and 
![image](https://user-images.githubusercontent.com/24397282/147105511-f6f4f1ba-239f-43b5-88ab-653473106e27.png)

chris-cedrone-cengage committed:
feat: pR updates

Adding in font size and color customizations along with updates to tests and docs to reflect
modifications.

chris-cedrone-cengage committed:
feat: adding changeset

aaronmaturen reviewed (changes requested):
https://css-tricks.com/quoting-in-html-quotations-citations-and-blockquotes/#hey-what-about-the-figure-element

we should probably use `figure` `blockquote` and `figcaption` for these...

chris-cedrone-cengage committed:
feat: adding in appropriate wrappers per Aaron

chris-cedrone-cengage committed:
feat: updates per feedback

chris-cedrone-cengage committed:
feat: updating doc copy

rcgillet commented on packages/react-magma-dom/src/components/BlockQuote/BlockQuote.test.js:
```diff
@@ -0,0 +1,137 @@
+const TEXT = 'Test Text';
+
+describe('BlockQuote', () => {
+  it('should render the visually hidden component', () => {
```
test description doesn't make sense.

rcgillet commented on packages/react-magma-dom/src/components/BlockQuote/BlockQuote.tsx:
```diff
@@ -0,0 +1,67 @@
+      <InverseContext.Provider value={{ isInverse }}>
+        <StyledBlockQuote
+          borderStyle={
+            magma.colors[borderStyle] ? magma.colors[borderStyle] : borderStyle
```
```suggestion
            magma.colors[borderStyle] || borderStyle
```

rcgillet commented on packages/react-magma-dom/src/components/BlockQuote/BlockQuoteItem.tsx:
```diff
@@ -0,0 +1,109 @@
+    <StyledBlockQuoteItem
+      as={hasAttribution ? 'figcaption' : 'blockquote'}
+      color={
+        hasAttribution
+          ? color || TypographyColor.subdued
+          : color || TypographyColor.default
```
```suggestion
        color || hasAttribution ? TypographyColor.subdued : TypographyColor.default
```

rcgillet commented on packages/react-magma-dom/src/components/BlockQuote/BlockQuoteItem.tsx:
```diff
@@ -0,0 +1,109 @@
+      isInverse={isInverse}
+      ref={ref}
+      data-testid={props.testId}
+      visualStyle={
+        hasAttribution
+          ? visualStyle || TypographyVisualStyle.bodyMedium
+          : visualStyle || TypographyVisualStyle.bodyLarge
```
```suggestion
      visualStyle={
        visualStyle || hasAttribution ? TypographyVisualStyle.bodyMedium : TypographyVisualStyle.bodyLarge
```

chris-cedrone-cengage committed:
feat: updates per PR, also caught spelling error in docs, amended

chris-cedrone-cengage committed:
feat: package lock

chris-cedrone-cengage committed:
feat: package lock fix

chris-cedrone-cengage committed:
feat: :(

chris-cedrone-cengage committed:
feat: package lock merge

chris-cedrone-cengage committed:
feat: package lock fix?

rcgillet committed:
Merge branch 'dev' into feature/blockquote

rcgillet committed:
test(blockquote): use custom jest-axe
